src/main/java/com/icthh/xm/ms/otp/service/impl/SpecLimitValidationServiceImpl.java

```diff
@@ -44,10 +44,10 @@ public void validateSpecificationLimit(final OneTimePasswordDto oneTimePasswordD
                 .map(validators::get)
                 .ifPresent(validator -> validator.validate(oneTimePasswordDto, otpSpec));
         } catch (InvalidPasswordException originException) {
-            if (otpSpec.getDiscloseCheckErrors()) {
```

Comment:
could "otpSpec" be null-value? why you didn't use ofNullable?

Reply:
that's impossible. if `otpSpec` is null, no exception would be thrown here. 

by the way, if for some reason spec is not found, IllegalArgumentException would be thrown here: com.icthh.xm.ms.otp.service.OtpSpecService#getOtpTypeSpec